server/neptune/gateway/event/source/file_fetcher.go

```diff
@@ -0,0 +1,66 @@
+package source
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"github.com/google/go-github/v45/github"
+	"github.com/palantir/go-githubapp/githubapp"
+	"github.com/runatlantis/atlantis/server/events/models"
+	"net/http"
+)
+
+type FileFetcher interface {
+	GetModifiedFilesFromCommit(ctx context.Context, repo models.Repo, sha string, installationToken int64) ([]string, error)
+}
+
+type GithubFileFetcher struct {
+	ClientCreator githubapp.ClientCreator
+}
+
+func (g *GithubFileFetcher) GetModifiedFilesFromCommit(ctx context.Context, repo models.Repo, sha string, installationToken int64) ([]string, error) {
+	var files []string
+	nextPage := 0
+	for {
+		opts := github.ListOptions{
+			PerPage: 300,
+		}
+		if nextPage != 0 {
+			opts.Page = nextPage
+		}
+		client, err := g.ClientCreator.NewInstallationClient(installationToken)
+		repositoryCommit, resp, err := client.Repositories.GetCommit(ctx, repo.Owner, repo.Name, sha, &opts)
+		if err != nil {
+			return files, err
+		}
+		if resp.StatusCode != http.StatusOK {
+			return nil, fmt.Errorf("error fetching repository contents: %s", resp.Status)
```

Comment:
Not sure if there's a specific convention around this but why are we retuning an empty list when err is not nil, and nil when the statusCode is not OK? Should we just return an empty list for both the scenarios instead?